cartographer/io/internal/mapping_state_serialization.cc

```diff
@@ -88,8 +88,10 @@ void SerializeSubmaps(
         submap_id_data.id.trajectory_id);
     submap_proto->mutable_submap_id()->set_submap_index(
         submap_id_data.id.submap_index);
-    submap_id_data.data.submap->ToProto(submap_proto,
-                                        /*include_probability_grid_data=*/true);
+    submap_id_data.data.submap->ToProto(
+        submap_proto,
+        /*include_probability_grid_data=*/submap_id_data.data.submap
+            ->finished());
```

Comment:
@cschuet Could you clarify the reason for this change? or why is it named a fix PR?
With this change the current state is not serialized but modified by leaving out data. 
This line also cases issue #1343.